ksql-engine/src/main/java/io/confluent/ksql/function/InternalFunctionRegistry.java

```diff
@@ -233,21 +233,39 @@ private void addMathFunctions() {
           AbsKudf.class));
 
       addBuiltInFunction(KsqlFunction.createLegacyBuiltIn(
-          Schema.OPTIONAL_FLOAT64_SCHEMA,
+          Schema.OPTIONAL_INT64_SCHEMA,
```

Comment:
I think this change would break backwards compatibility...

If customers already have a query running that uses the old code, when then upgrade the server will restart their query using the new code, but that will change the type of the result, which may well be incompatible.

We don't really have a way to version such UDFs to allow the return value to change between versions yet.

The _safe_ option is probably to stick with the old _incorrect_ return type for now. (Unless anyone has any ideas?) And just raise a Github issue to track, mentioning that just changing the return type would be a breaking change, so what we need first is a way to version UDFs and their result schemas.

Reply:
Yes, I was wondering  to correct it or not. The issue https://github.com/confluentinc/ksql/issues/2583 highlights the fact that FLOOR(3.6456) = 3 so not 3.0. I would recommend  to tackle it in separate thread.